src/js/directives/highlights.js

```diff
@@ -0,0 +1,138 @@
+var rangy = require('rangy');
+var _ = require('lodash');
+var $ = require('jquery');
+
+module.exports = function (app) {
+
+  // returns all leaf text nodes that are descendants of node or are node
+  var getTextNodes = function (node) {
+    if (!node) return [];
+    if (node.nodeType === Node.TEXT_NODE) return [node];
+    
+    // process childs
+    var nodes = [];
+    $(node).contents().each(function (index, el) {
+      nodes = nodes.concat(getTextNodes(el));
```

Comment:
`.push()`? [1] suggests that `concat` is faster, but then again we're doing this in-place 
[1] http://jsperf.com/javascript-array-concat-vs-push
[2] http://stackoverflow.com/questions/351409/appending-to-array